DataFormats/TrackerRecHit2D/interface/VectorHit.h

```diff
@@ -0,0 +1,168 @@
+#define DataFormats_TrackerRecHit2D_VectorHit_h
+
+/** \class VectorHit
+ *
+ * 4-parameter RecHits for Phase2 Tracker (x,y, dx/dz, dy/dz)
+ *
+ * $Date: 2015/03/30 $
+ * \author Erica Brondolin
+ *
+ */
+
+#include "DataFormats/TrackerRecHit2D/interface/BaseTrackerRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+#include "DataFormats/TrackerRecHit2D/interface/OmniClusterRef.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrackingRecHit/interface/KfComponentsHolder.h"
+
+#include "DataFormats/TrackerRecHit2D/interface/TkCloner.h"
+
+class VectorHit final : public BaseTrackerRecHit {
+public:
+  typedef OmniClusterRef::Phase2Cluster1DRef ClusterRef;
+
+  VectorHit() : thePosition(), theDirection(), theCovMatrix(), theDimension(0) { setType(bad); }
+
+  VectorHit(const VectorHit& vh);
+
+  VectorHit(const GeomDet& idet,
+            const LocalPoint& posInner,
+            const LocalVector& dir,
+            const std::array<std::array<float, 4>, 4> covMatrix,
+            const float chi2,
+            OmniClusterRef const& lower,
+            OmniClusterRef const& upper,
+            const float curvature,
+            const float curvatureError,
+            const float phi);
+
+  VectorHit(const GeomDet& idet,
+            const VectorHit2D& vh2Dzx,
+            const VectorHit2D& vh2Dzy,
+            OmniClusterRef const& lower,
+            OmniClusterRef const& upper,
+            const float curvature,
+            const float curvatureError,
+            const float phi);
+
+  ~VectorHit() override;
+
+  VectorHit* clone() const override { return new VectorHit(*this); }
+  RecHitPointer cloneSH() const override { return std::make_shared<VectorHit>(*this); }
+
+  bool sharesInput(const TrackingRecHit* other, SharedInputType what) const override;
+  bool sharesClusters(VectorHit const& h1, VectorHit const& h2, SharedInputType what) const;
+
+  // Parameters of the segment, for the track fit
+  // For a 4D segment: (dx/dz,dy/dz,x,y)
+  bool hasPositionAndError() const override {
+    //if det is present pose&err are available as well.
+    //if det() is not present (null) the hit ihas been read from file and not updated
+    return det();
+  };
+
+  AlgebraicVector parameters() const override;
+  void getKfComponents(KfComponentsHolder& holder) const override { getKfComponents4D(holder); }
+  void getKfComponents4D(KfComponentsHolder& holder) const;
+
+  // returning methods
+  LocalPoint localPosition() const override { return thePosition; }
+  virtual LocalVector localDirection() const { return theDirection; }
+  const std::array<std::array<float, 4>, 4> parametersErrorPlain() const;
```

Comment:
The method is needed to avoid creating `AlgebraicSymMatrix44` in the copy constructor and is currently only used for this purpose. To the outside world there is still the other `parametersError` method that creates and returns the `parametersError` object.

Reply:
the copy constructor should be able to use `vh.theCovMatrix`

Reply:
Function is gone.